libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/alert/IncidentAlert.kt

```diff
@@ -0,0 +1,251 @@
+            this.alertGeometry = alertGeometry
+        }
+
+        /**
+         * Unique identifier for incident. Must be **non-null**.
+         */
+        fun incidentId(incidentId: String): Builder = apply {
+            this.incidentId = incidentId
+        }
+
+        /**
+         * One of [IncidentAlertType].
+         */
+        fun incidentType(incidentType: Int): Builder = apply {
+            this.incidentType = incidentType
+        }
+
+        /**
+         * Severity level of incident.
+         */
+        fun impact(@IncidentAlertImpact.Impact impact: String?): Builder = apply {
+            this.impact = impact
+        }
+
+        /**
+         * Quantitative descriptor of congestion.
+         */
+        fun incidentCongestion(incidentCongestion: IncidentCongestion?): Builder = apply {
+            this.incidentCongestion = incidentCongestion
+        }
+
+        /**
+         * **True** if road is closed and no possibility to pass through there. **False**
+         * otherwise.
+         */
+        fun isClosed(isClosed: Boolean): Builder = apply {
+            this.isClosed = isClosed
+        }
+
+        /**
+         * Time the incident was created/updated in ISO8601 format. Incident can be
+         * created/updated before the incident.
+         */
+        fun creationTime(creationTime: Date?): Builder = apply {
+            this.creationTime = creationTime
+        }
+
+        /**
+         * Start time of the incident in ISO8601 format.
+         */
+        fun startTime(startTime: Date?): Builder = apply {
+            this.startTime = startTime
+        }
+
+        /**
+         * End time of the incident in ISO8601 format.
+         */
+        fun endTime(endTime: Date?): Builder = apply {
+            this.endTime = endTime
+        }
+
+        /**
+         * Human-readable description of the incident suitable for displaying to the users.
+         */
+        fun description(description: String?): Builder = apply {
+            this.description = description
+        }
+
+        /**
+         * Sub-type of the incident.
+         */
+        fun subType(subType: String?): Builder = apply {
+            this.subType = subType
+        }
+
+        /**
+         * Sub-type-specific description.
+         */
+        fun subTypeDescription(subTypeDescription: String?): Builder = apply {
+            this.subTypeDescription = subTypeDescription
+        }
+
+        /**
+         * AlertC codes.
+         *
+         * @see <a href="https://www.iso.org/standard/59231.html">AlertC</a>
+         */
+        fun alertcCodes(alertcCodes: List<Int>?): Builder = apply {
+            this.alertcCodes = alertcCodes
+        }
+
+        /**
+         * Build the object instance.
+         */
+        fun build(): IncidentAlert {
+            val id = incidentId
+                ?: throw IllegalStateException("IncidentAlert, incidentId must be non-null")
```

Comment:
If we're throwing, then it should definitely be a constructor parameter instead.